src/schema.rs

```diff
@@ -125,13 +126,26 @@ impl TryCmd {
         id: Option<&str>,
         stdout: Option<&crate::Data>,
         stderr: Option<&crate::Data>,
+        exit: Option<ExitStatus>,
     ) -> Result<(), crate::Error> {
         if let Some(ext) = path.extension() {
             if ext == std::ffi::OsStr::new("toml") {
                 assert_eq!(id, None);
 
                 overwrite_toml_output(path, id, stdout, "stdout", "stdout")?;
                 overwrite_toml_output(path, id, stderr, "stderr", "stderr")?;
+
+                if let Some(code) = exit.and_then(|status| status.code()) {
+                    let raw = std::fs::read_to_string(path)
+                        .map_err(|e| format!("Failed to read {}: {}", path.display(), e))?;
+                    let mut doc = raw
+                        .parse::<toml_edit::Document>()
+                        .map_err(|e| format!("Failed to read {}: {}", path.display(), e))?;
+                    doc["status"] = toml_edit::value(toml_edit::InlineTable::default());
+                    doc["status"]["code"] = toml_edit::value(code as i64);
+                    std::fs::write(path, doc.to_string())
+                        .map_err(|e| format!("Failed to write {}: {}", path.display(), e))?;
```

Comment:
Looks like this unconditionally uses the code.  I feel like we should preserve the users existing style.